src/map/atcommand.c

```diff
@@ -5479,6 +5479,46 @@ ACMD(divorce)
 }
 
 /*==========================================
+ * @adopt by Aleos
+ * Usage: @adopt <char name>
+ *------------------------------------------*/
+
+ACMD(adopt)
+{
+	struct map_session_data *b_sd;
+	enum adopt_responses response;
+
+	memset(atcmd_output, '\0', sizeof(atcmd_output));
+	memset(atcmd_player_name, '\0', sizeof(atcmd_player_name));
+
+	if (!message[0] || sscanf(message, "%23[^\n]", atcmd_player_name) < 1) {
+		sprintf(atcmd_output, msg_fd(fd, 435), command); // Please enter a player name (usage: %s <char name>).
+		clif->message(fd, atcmd_output);
+		return false;
+	}
+
+	if ((b_sd = map->nick2sd((char *)atcmd_player_name)) == NULL) {
```

Comment:
Can you remove the typecast from here? The variable is already a pointer to char... Typecasts can easily (accidentally) hide other issues that the compiler would normally warn about, I'd not add them where they aren't unnecessary